### PR TITLE
 bug(backend) fix endpoints issues

### PR DESCRIPTION
**Pull request by jo-rdan**
#### What does this PR do?
It fixes endpoints issues in the project
#### Description of Task to be completed?
- change response structure to adhere to the ADC response spec
- change create session request endpoint to restrict the number of times the user requests the session depending on the questions he asked
 change reviews endpoint to restrict the number of times a user reviews the same mentor
- add unit tests to test any uncovered line
#### How should this be manually tested?
- First clone this repository to your local machine, and install all the packages 
- Open the cloned folder ans start the server by typing **npm start** 

**endpoint to create a session request**
- In POSTMAN start a POST request on the following endpoint **/api/v1/sessions** and hit send more than once and it should respond with a relevant error message
- Change the question field and create a request and should respond with a **201** HTTP status code 

**endpoint to review a mentor**
- In POSTMAN start a POST request on the following endpoint **/api/v1/sessions/:id(id is the id of the session request)/review** and hit send more than once and it should respond with a relevant error message
- Change the question field and create a request and should respond with a **200** HTTP status code 
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#168284746](https://www.pivotaltracker.com/story/show/168284746)
#### Screenshots (if appropriate)
**create session request**
image

**review mentor**
image
image

### CODE REVIEW

File: server/control/sessionControl.js [hunk 1]
```diff
@@ -55,20 +59,23 @@ class SessionController {
   static reviewMentor(req, res) {
     const { score, remark } = req.body;
     const isSession = sessions.find(f => f.sessionId == req.params.id);
```
Comment (ekumamait) [inline server/control/sessionControl.js]: We talked about naming variables with more meaninful names not `f` or `p`

File: server/tests/userTests.js [hunk 2]
```diff
@@ -781,6 +826,17 @@ describe('Review Tests', () => {
        done();
      })
  }));
+ it('should not be able to review session when session is already reviewed', (done => {
+   const id = 3;
+   chai.request(app).post(`/api/v1/sessions/${id}/review`).set('x-token', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwiZW1haWwiOiJrYXlpbmFtdXJhQGdtYWlsLmNvbSIsImlzQWRtaW4iOmZhbHNlLCJpYXQiOjE1Njc2NTQ2OTB9.P-H5pIYBfhqT_Thr6TOk_6qCkIUgMVIlKaYZFGf-N4Q').send({
```
Comment (ekumamait) [inline server/tests/userTests.js]: Let's keep our tests DRY by importing a `mockData` for tests that has `Tokens`

File: server/tests/userTests.js [hunk 3]
```diff
@@ -395,8 +413,18 @@ describe('Authentication tests', () => {
         done();
       });
   });
+  it('any error', (done) => {
+    chai.request(app).post('/api/v1/auth/signin').send({
```
Comment (ekumamait) [inline server/tests/userTests.js]: DRY testing

**Review by ekumamait — COMMENTED** [review]
Please take a look at the inline comments

File: server/control/sessionControl.js (hunk 1 above)
Comment (jo-rdan) [reply server/control/sessionControl.js]: yes, I was going to do that on a separate branch along with the dry testing issue or should I just do it on this branch and commit again

**Pull request merged**